src/index.c

```diff
@@ -2645,11 +2650,8 @@ static int parse_index(git_index *index, const char *buffer, size_t buffer_size)
 	while (buffer_size > INDEX_FOOTER_SIZE) {
 		size_t extension_size;
 
-		extension_size = read_extension(index, buffer, buffer_size);
-
-		/* see if we have read any bytes from the extension */
-		if (extension_size == 0) {
-			error = index_error_invalid("extension is truncated");
+		error = read_extension(&extension_size, index, buffer, buffer_size);
+		if (error < 0) {
```

Comment:
Minor nit: I'd reformat this to our usual style of

```
if ((error = read_extension(&extension_size, index, buffer, buffer_size)) < 0)
    goto done;
```

Please only consider this if you are already rerolling :)